addons/event_sale/models/event_ticket.py

```diff
@@ -1,9 +1,81 @@
 from odoo.exceptions import ValidationError, UserError
 
+_logger = logging.getLogger(__name__)
+
+
+class EventTemplateTicket(models.Model):
+    _inherit = 'event.type.ticket'
+
+    def _default_product_id(self):
+        return self.env.ref('event_sale.product_product_event', raise_if_not_found=False)
+
+    # product
+    product_id = fields.Many2one(
+        'product.product', string='Product', required=True, domain=[("event_ok", "=", True)],
+        default=_default_product_id)
+    price = fields.Float(string='Price', digits='Product Price')
+    price_reduce = fields.Float(string="Price Reduce", compute="_compute_price_reduce", digits='Product Price')
+
+    @api.depends('product_id')
+    def _compute_price_reduce(self):
+        for record in self:
+            product = record.product_id
+            discount = product.lst_price and (product.lst_price - product.price) / product.lst_price or 0.0
+            record.price_reduce = (1.0 - discount) * record.price
+
+    @api.depends('product_id.active')
+    def _compute_sale_available(self):
+        inactive_product_tickets = self.filtered(lambda ticket: not ticket.product_id.active)
+        for ticket in inactive_product_tickets:
+            ticket.sale_available = False
+        super(EventTemplateTicket, self - inactive_product_tickets)._compute_sale_available()
+
+    @api.onchange('product_id')
+    def _onchange_product_id(self):
+        self.price = self.product_id.list_price or 0
+
+    def _init_column(self, column_name):
+        if column_name != "product_id":
+            return super(EventTemplateTicket, self)._init_column(column_name)
+
+        # fetch void columns
+        self.env.cr.execute("SELECT id FROM %s WHERE product_id IS NULL" % self._table)
+        ticket_type_ids = self.env.cr.fetchall()
+        if not ticket_type_ids:
+            return
+
+        # update existing columns
+        _logger.debug("Table '%s': setting default value of new column %s to unique values for each row",
+                      self._table, column_name)
+        default_event_product = self.env.ref('event_sale.product_product_event', raise_if_not_found=False)
+        if default_event_product:
+            product_id = default_event_product.id
+        else:
+            product_id = self.env['product.product'].create({
+                'name': 'Generic Registration Product',
+                'list_price': 0,
+                'standard_price': 0,
+                'type': 'service',
+                'default_code': 'EVENT_REG',
+                'type': 'service',
+            }).id
+            self.env['ir.model.data'].create({
+                'name': 'product_product_event',
+                'module': 'event_sale',
+                'model': 'product.product',
+                'res_id': product_id,
+            })
+        self.env.cr._obj.execute(
+            'UPDATE %s SET product_id = %%s WHERE id = %%s;' % (self._table),
+            (product_id, tuple(ticket_type_ids))
```

Comment:
Could use "template strings" like XMO explained to us last time \o/

Reply:
Do you remember where we did it ?

Reply:
I used it yesterday to implement the "short survey route" thing.

For your use case, should be something like:

```py
f'UPDATE {self._table} SET product_id = %s WHERE id = %s;'
```

I try to use this whenever I heave to deal with "%%s" now, much more readable IMO.

Reply:
Updated.